Sources/App/Core/Search.swift

```diff
@@ -175,7 +194,34 @@ enum Search {
             .column(keywords)
             .column(hasDocs)
             .column(nullInt, as: levenshteinDist)
+            .column(ts_rank(vector: tsvector, query: tsquery), as: tsrankvalue)
             .from(searchView)
+            .from(plainto_tsquery(mergedTerms), as: tsquery)
+            // SIMPLE VERSION: .from(to_tsvector(coalesce(packageName, emptyString)), as: tsvector)
+            // RAW_VERSION:
+            .from(SQLRaw("setweight(to_tsvector(coalesce(summary, '') || coalesce(package_name, '') || coalesce(ARRAY_TO_STRING(keywords, ' '), '')), 'A')"), as: tsvector)
+        // My attempt at making this into something more SQLKit-ish isn't working
+        // at the moment, something to do with a syntax error around the concatenate mechanism...
+        // The || operator is working on concatenating strings, NOT vectors, so all the SQLKit stuff
+        // I wrote may be absolutely wrong. The above raw at least binds it all into a single weighting set,
+        // and seems to 'function'.
+        
+        // The example from the Postgres docs for individually weighting string values from a structured
+        // document (https://www.postgresql.org/docs/current/textsearch-controls.html)
+        //
+//        UPDATE tt SET ti =
+//            setweight(to_tsvector(coalesce(title,'')), 'A')    ||
+//            setweight(to_tsvector(coalesce(keyword,'')), 'B')  ||
+//            setweight(to_tsvector(coalesce(abstract,'')), 'C') ||
+//            setweight(to_tsvector(coalesce(body,'')), 'D');
+        
+//            .from(vecMerge(
+//                to_tsvectorB(coalesce(keywords, emptyString)), // keyword vector as weight B
+//                    vecMerge(
+//                        to_tsvectorA(coalesce(packageName, emptyString)), // package name vector  as weight A
+//                        to_tsvector(coalesce(summary, emptyString)) // summary vector as weight D
+//                    )
+//                ), as: tsvector)
```

Comment:
We'd definitely want to either find an SQLKit way to express it with the existing bind variable mechanisms or at least make sure the `SQLRaw` way isn't open to SQL injection. I suspect it wouldn't be but I'm not sure.

I'm going to be a bit pressed for time in the next week or two but hopefully I'll find some time to take a look!

Reply:
I suspect my error is in the SQL, I need to iterate and make sure I was doing something legit, but I'm confident that we can make it work with SQLKit once I nail down the correct underlying SQL.

I wasn't 100% sure about how to properly form a SQLFunction with just two, three, or four parameters - so in the case of a SQLKit version of `tsquery`, I made it explicitly named parameters. The tsvector concatenation gets a little more awkward seen in swift code though.